src/main/java/com/minecolonies/coremod/util/BlockUtils.java

```diff
@@ -400,4 +400,37 @@ public static boolean isPathBlock(final Block block)
     {
         return block == Blocks.GRAVEL || block == Blocks.STONEBRICK;
     }
+
+    /**
+     * Getter for a Block from an itemStack.
+     * @param stack the incoming itemStack
+     * @return the outgoing block
+     */
+    public static Block getBlockFromItemStack(final ItemStack stack)
+    {
+        Block block = Block.getBlockFromItem(stack.getItem());
+
+        if(block != null)
+        {
+            return block;
+        }
+
+        if(stack.getItem() instanceof ItemDoor)
+        {
+            return stack.getItem() == Items.OAK_DOOR ? Blocks.OAK_DOOR
+                    : stack.getItem() == Items.SPRUCE_DOOR ? Blocks.SPRUCE_DOOR
+                            : stack.getItem() == Items.BIRCH_DOOR ? Blocks.BIRCH_DOOR
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS864)